airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -538,4 +536,117 @@ protected boolean verifyCursorColumnValues(final JdbcDatabase database, final St
     return !nullValExist;
   }
 
+  @Override
+  protected void estimateFullRefreshSyncSize(final JdbcDatabase database,
+                                             final ConfiguredAirbyteStream configuredAirbyteStream) {
+    try {
+      final String schemaName = configuredAirbyteStream.getStream().getNamespace();
+      final String tableName = configuredAirbyteStream.getStream().getName();
+      final String fullTableName =
+          getFullyQualifiedTableNameWithQuoting(getQuoteString(), schemaName, tableName);
+
+      final List<JsonNode> tableEstimateResult = getFullTableEstimate(database, fullTableName);
+
+      final long syncRowCount = tableEstimateResult.get(0).get(ROW_COUNT_RESULT_COL).asLong();
+      final long syncByteCount = tableEstimateResult.get(0).get(TOTAL_BYTES_RESULT_COL).asLong();
+
+      // Here, we double the bytes estimate to account for serialization. Perhaps a better way to do this
+      // is to
+      // read a row and Stringify it to better understand the accurate volume of data sent over the wire.
+      // However, this approach doesn't account for different row sizes.
+      AirbyteTraceMessageUtility.emitEstimateTrace(2 * syncByteCount, Type.STREAM, syncRowCount, tableName, schemaName);
+      LOGGER.info("estimate for table: " + fullTableName + " : {total_bytes:" + syncByteCount + ", row_count : " + syncRowCount);
+    } catch (final SQLException e) {
+      throw new ConfigErrorException("Error occurred while attempting to estimate sync size", e);
+    }
+  }
+
+  @Override
+  protected void estimateIncrementalSyncSize(final JdbcDatabase database,
+                                             final ConfiguredAirbyteStream configuredAirbyteStream,
+                                             final CursorInfo cursorInfo,
+                                             final PostgresType cursorFieldType) {
+    try {
+      final String schemaName = configuredAirbyteStream.getStream().getNamespace();
+      final String tableName = configuredAirbyteStream.getStream().getName();
+      final String fullTableName =
+          getFullyQualifiedTableNameWithQuoting(getQuoteString(), schemaName, tableName);
+
+      final List<JsonNode> tableEstimateResult = getFullTableEstimate(database, fullTableName);
+
+      final long tableRowCount = tableEstimateResult.get(0).get(ROW_COUNT_RESULT_COL).asLong();
+      final long tableByteCount = tableEstimateResult.get(0).get(TOTAL_BYTES_RESULT_COL).asLong();
```

Comment:
Unless I'm mistaken, these two values are only created once. Is it valuable to have these values logged so that they are persisted within the logs? 

```java
Logger.log(String.format("Total table row count: %s, Total table byte count: %s", tableRowCount, tableByteCount));
```

Reply:
I've logged them below